cetmix_tower_server/models/cx_tower_reference_mixin.py

```diff
@@ -126,14 +135,38 @@ def create(self, vals_list):
         Returns:
             Records: The created record(s).
         """
-        if not self._context.get("reference_mixin_override"):
+
+        if vals_list and not self._context.get("reference_mixin_override"):
+            # Check if we need to populate references based on parent record
+            auto_generate_settings = self.GENERATE_FROM_RELATED.get(self._name)
+            if auto_generate_settings:
+                parent_model, relation_field, suffix = auto_generate_settings
+                vals_list = self._pre_populate_references(
+                    parent_model, relation_field, vals_list, suffix
+                )
+
+            # Fix or create references
             for vals in vals_list:
-                if vals.get("name"):
-                    vals["name"] = vals["name"].strip()
+                if not vals:
+                    continue
```

Comment:
Why do we need this check? the name field is required, how can it be empty? In any case, you also have further checks on the name and reference fields. this check is unnecessary.

Reply:
Because `name` is required in mixin. However if some model inherits this mixin and makes `name` a non-required field this may lead to an unpredictable behaviour.